### ISSUE TITLE
Max Priority gas price does not work

### ISSUE THREAD

**Issue by robinsdan** [foundry-rs/foundry#7486]
### Component

Forge

### Have you ensured that all of these are up to date?

- [ ] Foundry
- [ ] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (7545c7a2 2024-03-12T03:19:04.521042379Z)

### What command(s) is the bug in?

forge script

### Operating System

Linux

### Describe the bug

I use `forge script -v --priority-gas-price 0.05gwei --with-gas-price 20gwei`,  but the transaction will set the priority gas to 20GWEI. 

```
Gas Fees:
Base: 15.038300915 Gwei |Max: 20 Gwei |Max Priority: 20 Gwei
```

**Comment by mattsse** [comment]
looks like priority gas price is overwritten by gas price

      --with-gas-price <PRICE>
          Gas price for legacy transactions, or max fee per gas for EIP1559 transactions

@klkvr I think simply changing the order in which those are set should fix this

**Comment by klkvr** [comment]
Will be addressed in scope of #7106
